subsys/bluetooth/audio/bap_broadcast_assistant.c

```diff
@@ -503,9 +532,10 @@ static uint8_t notify_handler(struct bt_conn *conn,
 			 */
 			inst->long_read_handle = handle;
 
-			if (!inst->busy) {
+			if (!atomic_test_bit(inst->flags, BAP_BA_FLAG_BUSY)) {
 				net_buf_simple_add_mem(&att_buf, data, length);
```

Comment:
Isn't this racy? As soon as `atomic_test_bit()` returns you might get preempted and the flag could get set. Or it could get set right after the `net_buf_simple_add_mem()` call. Point is, just testing for the bit doesn't really tell you anything else except the state of the BUSY flag at that exact instance but not anywhere else.

Reply:
That is a fair point. 

Looking through the code and the usage of `att_buf`, I've determined that the `att_buf` is always reset before and after each use. 

This means that if another thread starts another procedure between this line and the `if (atomic_test_and_set_bit(inst->flags, BAP_BA_FLAG_BUSY)) {` in `long_bap_read`, then the memory is dropped (which is fine as we will just start reading from `0` instead of `length` as the offset). 

So we will either 
A) If it becomes busy between here and `long_bap_read` then we discard the notification that requires a long read, but will retry again later
B) Else we will start a long read process where we are setting the busy flag in this thread